### ISSUE TITLE
InfoBar action button loading state issue

### ISSUE THREAD

**Issue by samyakj-eightfold** [EightfoldAI/octuple#841]
If the loading prop is set for action button in InfoBar then the loader overlaps the button text and not replace it.
Sandbox [link](https://codesandbox.io/p/sandbox/info-bar-forked-5cgtcv?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clwkloau900063b6lsyo5p09h%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clwkloau900023b6lwx8xe0mn%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clwkloau900033b6lizwcwvbp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clwkloau900053b6lv2pt315i%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clwkloau900023b6lwx8xe0mn%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwkloau900013b6l4739kx7x%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clwkloau900023b6lwx8xe0mn%2522%252C%2522activeTabId%2522%253A%2522clwkloau900013b6l4739kx7x%2522%257D%252C%2522clwkloau900053b6lv2pt315i%2522%253A%257B%2522id%2522%253A%2522clwkloau900053b6lv2pt315i%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clwkloau900043b6lv3rhfmy5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clwkloau900043b6lv3rhfmy5%2522%257D%252C%2522clwkloau900033b6lizwcwvbp%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clwkloau900033b6lizwcwvbp%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

<img width="1920" alt="Screenshot 2024-05-24 at 5 06 19 PM" src="https://github.com/EightfoldAI/octuple/assets/77293737/af9b6f95-90ad-4827-92cf-34ffd051e897">

**Comment by dkilgore-eightfold** [comment]
@samyakj-eightfold For now, set the button text based upon the loading state

**Comment by samyakj-eightfold** [comment]
> @samyakj-eightfold For now, set the button text based upon the loading state

@dkilgore-eightfold Yes, an empty string was used to hide the text behind the loader but the loader appears at the end differently.
Refer [ApplicationNoteCapability](https://github.com/EightfoldAI/vscode/blob/master/www/react/src/apps/careerHub/projectMarketplaceManagerV2/components/ProjectApplyV2/ApplicationNoteCapability.js#L40)

<img width="1920" alt="Screenshot 2024-05-30 at 1 08 41 AM" src="https://github.com/EightfoldAI/octuple/assets/77293737/0bb023b8-1bb8-4e73-88e6-e777a533d5a4">
